public/app/features/dashboard/panel_editor/VizTypePickerPlugin.tsx

```diff
@@ -135,6 +136,13 @@ interface PanelPluginBadgeProps {
 const PanelPluginBadge: React.FC<PanelPluginBadgeProps> = ({ plugin }) => {
   const display = getPanelStateBadgeDisplayModel(plugin);
 
+  if (
+    plugin.signature &&
+    plugin.signature !== PluginSignatureStatus.valid &&
+    plugin.signature !== PluginSignatureStatus.internal
```

Comment:
Similar logic to the one in `DataSourcePicker.tsx` on lines 93-95. I'd consider extracting this into a function.